examples/graph/tool_requires/using_protobuf.rst

```diff
@@ -0,0 +1,230 @@
+.. _examples_graph_tool_requires_protobuf:
+
+Using the same requirement as a requires and as a tool_requires
+=====================================================================
+
+There are libraries which could behave as a library and as a tool requirement, e.g., `protobuf <https://github.com/conan-io/conan-center-index/tree/master/recipes/protobuf>`__
+Those libraries normally contains headers/sources that compound the library itself, and, perhaps, an extra tools
+(compilers, shell scripts, etc.). Both parts are used in a different contexts, let's think of this scenario using
+*protobuf* for instance:
+
+* I want to create a library which includes a compiled protobuf message. The protobuf compiler (build context)
+  needs to be invoked at build time, and the library with the compiled *.pb.cc* file needs to be linked against
+  the protobuf library (host context).
+
+
+Given that, we should be able to use protobuf in build/host context in the same Conan recipe. Basically, your package
+should look like:
+
+.. code-block:: python
+
+    def requirements(self):
+        self.requires("protobuf/3.18.1")
+
+    def build_requirements(self):
+        self.tool_requires("protobuf/3.18.1")
+
+
+This is the way to proceed with any other library used in both contexts. Nonetheless, let's see a detailed example to see
+how the example looks like.
+
+Please, first, clone the sources to recreate this project, you can find them in the
+`examples2.0 repository <https://github.com/conan-io/examples2>`_ on GitHub:
+
+.. code-block:: shell
+
+    git clone https://github.com/conan-io/examples2.git
+    cd examples2/examples/graph/tool_requires/using_protobuf/myserver
+
+
+The structure of the project is the following:
+
+.. code-block:: text
+
+    ./
+    ├── conanfile.py
+    ├── CMakeLists.txt
+    ├── addressbook.proto
+    ├── host-armv8
+    ├── host-x86_64
```

Comment:
If these are specific to Mac, maybe call them ``apple-xxx``? Also avoid the "host" because one of them is for "build"?